WcaOnRails/app/assets/javascripts/competitions.js

```diff
@@ -3,6 +3,10 @@ onPage('competitions#edit, competitions#update, competitions#admin_edit, competi
     $('.wca-registration-options').toggle(this.checked);
   }).trigger('change');
 
+  $('input[name="competition[on_the_spot_registration]"]').on('change', function() {
+    $('.wca-on-the-spot-registration-options').toggle(this.checked);
+  }).trigger('change');
+
```

Comment:
I left this because I'm not sure if it's actually as it should be or not. It's an adapted copy of the competitior limit one just below.

Reply:
Oh, on further reflection, this looks reasonable to me. Sorry for the hassle!